src/Campaigns/Migrations/MigrateFormsToCampaignForms.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+namespace Give\Campaigns\Migrations;
+
+use DateTime;
+use Give\Campaigns\Models\Campaign;
+use Give\Campaigns\ValueObjects\CampaignStatus;
+use Give\Campaigns\ValueObjects\CampaignType;
+use Give\DonationForms\Models\DonationForm;
+use Give\DonationForms\ValueObjects\DonationFormStatus;
+use Give\Framework\Database\DB;
+use Give\Framework\Migrations\Contracts\Migration;
+
+/**
+ * @unreleased
+ */
+class MigrateFormsToCampaignForms extends Migration
+{
+    /**
+     * @inheritDoc
+     */
+    public static function id(): string
+    {
+        return 'migrate_forms_to_campaign_forms';
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public static function timestamp(): int
+    {
+        return strtotime('2024-08-21');
+    }
+
+    /**
+     * @unreleased
+     * @inheritDoc
+     */
+    public function run()
+    {
+        foreach(DonationForm::query()->getAll() ?? [] as $form) {
+            $this->createParentCampaignForDonationForm($form);
+        }
+    }
+
+    /**
+     * @unreleased
+     */
+    public function createParentCampaignForDonationForm(DonationForm $form)
+    {
+        $campaign = Campaign::create([
+            'pageId' => 0,
+            'type' => CampaignType::CORE(),
+            'title' => $form->title,
+            'shortDescription' => $form->settings->formExcerpt,
+            'longDescription' => $form->settings->description,
+            'logo' => $form->settings->designSettingsLogoUrl,
+            'image' => $form->settings->designSettingsImageUrl,
+            'primaryColor' => $form->settings->primaryColor,
+            'secondaryColor' => $form->settings->secondaryColor,
+            'goal' => (int) $form->settings->goalAmount,
+            'status' => $this->mapFormStatusToCampaignStatus($form->status),
+            'startDate' => new DateTime($form->settings->goalStartDate),
+            'endDate' => new DateTime($form->settings->goalEndDate),
+        ]);
+
+        DB::table('give_campaign_forms')
+            ->insert([
+                'form_id' => $form->id,
+                'campaign_id' => $campaign->id,
+            ]);
+    }
+
+   /**
+     * @unreleased
+     */
+    public function mapFormStatusToCampaignStatus(DonationFormStatus $status)
+    {
+        switch ($status) {
+            case DonationFormStatus::PUBLISHED():
+            case DonationFormStatus::UPGRADED(): // TODO: How do we handle upgraded, non-upgraded forms?
```

Comment:
This is a good question. If each form creates a campaign, we will still need an option to upgrade the v2 form. We can do that automatically during migration, but that's probably something people don't want.

Reply:
> We'll be including v2 forms.
> We're focusing on v3 forms first, so the process isn't completely fleshed out for v2.

Per Angela, v2 forms will be included in the migration process to campaigns, but there are still unknowns.

Cross-link: https://lw.slack.com/archives/C04SLRDD9CK/p1725462015102489?thread_ts=1725456372.298229&cid=C04SLRDD9CK